### PR TITLE
Dfcc 136 apply vs comp ui template for new child app

### PR DESCRIPTION
**Pull request by IanCrisp**
Solution created from VS Template for Comp UI Child apps - initial solution development

### CODE REVIEW

File: DFC.App.ContactUs/appsettings-template.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "Logging": {
+    "LogLevel": {
+      "Default": "Debug"
+    }
+  },
+  "AllowedHosts": "*",
+  "Configuration": {
+    "CosmosDbConnections": {
+      "ContentPages": {
+        "AccessKey": "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",
```
Comment (snipered2003) [inline DFC.App.ContactUs/appsettings-template.json:11]: remove app key

File: DFC.App.ContactUs.Data.UnitTests/Validation/ContentPageModelValidationTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+namespace DFC.App.ContactUs.Data.UnitTests.Validation
+{
+    [TestFixture]
+    [Category("AppValidation.Tests")]
+    public class ContentPageModelValidationTests
+    {
+        private const string GuidEmpty = "00000000-0000-0000-0000-000000000000";
+
+        [SetUp]
+        public void SetUp()
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Data.UnitTests/Validation/ContentPageModelValidationTests.cs:19]: Will remove

File: DFC.App.ContactUs.IntegrationTests/appsettings-template.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "Logging": {
+    "LogLevel": {
+      "Default": "Debug"
+    }
+  },
+  "AllowedHosts": "*",
+  "Configuration": {
+    "CosmosDbConnections": {
+      "ContentPages": {
+        "AccessKey": "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.IntegrationTests/appsettings-template.json:11]: This is the well-known public key for the Cosmos Emulator - not a security issue

File: DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            }
+
+            return models.Any() ? models : null;
+        }
+
+        public async Task<HttpStatusCode> UpsertAsync(T model)
+        {
+            await InitialiseDevEnvironment().ConfigureAwait(false);
+
+            var ac = new AccessCondition { Condition = model.Etag, Type = AccessConditionType.IfMatch };
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs:144]: Renamed to something suitable

File: DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+            var pk = new PartitionKey(model.PartitionKey);
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs:145]: Renamed to something suitable

File: DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+
+        public async Task<HttpStatusCode> DeleteAsync(Guid documentId)
+        {
+            var documentUri = CreateDocumentUri(documentId);
+
+            var model = await GetAsync(d => d.DocumentId == documentId).ConfigureAwait(false);
+
+            if (model != null)
+            {
+                var ac = new AccessCondition { Condition = model.Etag, Type = AccessConditionType.IfMatch };
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs:160]: Renamed to something suitable

File: DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+                var pk = new PartitionKey(model.PartitionKey);
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs:161]: Renamed to something suitable

File: DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+        {
+            try
+            {
+                await documentClient.ReadDocumentCollectionAsync(UriFactory.CreateDocumentCollectionUri(cosmosDbConnection.DatabaseId, cosmosDbConnection.CollectionId)).ConfigureAwait(false);
+            }
+            catch (DocumentClientException e)
+            {
+                if (e.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    var pkDef = new PartitionKeyDefinition
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs:200]: Renamed to something suitable

File: DFC.App.ContactUs/appsettings-template.json (hunk 1 above)
Comment (IanCrisp) [reply DFC.App.ContactUs/appsettings-template.json:11]: This is the well-known public key for the Cosmos Emulator - not a security issue

File: DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs [hunk 9]
```diff
@@ -0,0 +1,222 @@
...
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        private Uri CreateDocumentUri(Guid documentId)
+        {
+            return UriFactory.CreateDocumentUri(cosmosDbConnection.DatabaseId, cosmosDbConnection.CollectionId, documentId.ToString());
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Repository.CosmosDb/CosmosRepository.cs:219]: documentId is a Guid so I do not think it is necessary

File: DFC.App.ContactUs.Data/Attributes/LowerCaseAttribute.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+
+namespace DFC.App.ContactUs.Data.Attributes
+{
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public class LowerCaseAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Data/Attributes/LowerCaseAttribute.cs:13]: Uses the same name from the overridden base method's parameters

File: DFC.App.ContactUs.Data/Attributes/UrlPathAttribute.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+
+namespace DFC.App.ContactUs.Data.Attributes
+{
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public class UrlPathAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.ContactUs.Data/Attributes/UrlPathAttribute.cs:13]: Uses the same name from the overridden base method's parameters

**Pull request merged**